common/hexutil/hexutil_test.go

```diff
@@ -28,9 +28,10 @@ type marshalTest struct {
 }
 
 type unmarshalTest struct {
-	input   string
-	want    interface{}
-	wantErr error
+	input        string
+	want         interface{}
+	wantErr      error // if set, decoding must fail on any platform
+	wantErr32bit error // if set, decoding must fail on 32bit platforms (used for Uint tests)
```

Comment:
`uint` size depends on target arch, it's a fact of life. 

Let's say you have a (stdlib) API that expects a `uint` parameter. `hexutil.Uint` is nice if you want to decode a number from user input and use it with the API. The decoder does the range check for you.

The alternative would be to use `hexutil.Uint64` for decoding and then convert to `uint` when calling the API. To be safe against overflow, you'd need a check against `math.MaxUint`. Nobody would write those checks, so there'd just be a silent overflow.